internal/cortex/frontend/transport/handler.go

```diff
@@ -182,9 +166,14 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 		writeError(w, err)
 		queryString = f.parseRequestQueryString(r, buf)
```

Comment:
queryString is parsed twice in some situations. See the code snippet below. Can you make it only parse once? 
```
if shouldReportSlowQuery || f.cfg.QueryStatsEnabled {
		queryString = f.parseRequestQueryString(r, buf)
	}
```

Reply:
It would not be parsed twice since if err != nil, the method returns after reporting a failed query so it never checks if the query is a slow query and does not call `parseRequestQueryString` again.